pkg/operator/controller/ingress/controller.go

```diff
@@ -632,6 +625,44 @@ func setDefaultPublishingStrategy(ic *operatorv1.IngressController, platformStat
 	return false
 }
 
+// setDefaultProviderParameters mutates the given LoadBalancerStrategy by
+// defaulting its ProviderParameters field based on the defaults in the provided
+// ingress config object.
+func setDefaultProviderParameters(lbs *operatorv1.LoadBalancerStrategy, ingressConfig *configv1.Ingress) {
+	if lbs.ProviderParameters == nil {
+		lbs.ProviderParameters = &operatorv1.ProviderLoadBalancerParameters{}
+	}
+	if len(lbs.ProviderParameters.Type) == 0 {
+		// Infer the LB type from the cluster ingress config.
+		switch ingressConfig.Spec.LoadBalancer.Platform.Type {
+		case configv1.AWSPlatformType:
+			lbs.ProviderParameters.Type = operatorv1.AWSLoadBalancerProvider
+		}
+	}
+	switch lbs.ProviderParameters.Type {
+	case operatorv1.AWSLoadBalancerProvider:
+		defaultLBType := operatorv1.AWSClassicLoadBalancer
+		if p := ingressConfig.Spec.LoadBalancer.Platform; p.Type == configv1.AWSPlatformType && p.AWS != nil {
+			if p.AWS.Type == configv1.NLB {
+				defaultLBType = operatorv1.AWSNetworkLoadBalancer
+			}
+		}
+		if lbs.ProviderParameters.AWS == nil {
+			lbs.ProviderParameters.AWS = &operatorv1.AWSLoadBalancerParameters{}
+		}
+		if len(lbs.ProviderParameters.AWS.Type) == 0 {
+			lbs.ProviderParameters.AWS.Type = defaultLBType
+		}
+		switch lbs.ProviderParameters.AWS.Type {
+		case operatorv1.AWSClassicLoadBalancer:
+			if lbs.ProviderParameters.AWS.ClassicLoadBalancerParameters == nil {
+				lbs.ProviderParameters.AWS.ClassicLoadBalancerParameters = &operatorv1.AWSClassicLoadBalancerParameters{}
+			}
+		}
+	}
```

Comment:
Sorry, I need to revise my earlier suggestion.  `setDefaultPublishingStrategy` now assumes that `specLB.ProviderParameters.GCP` is set by `setDefaultProviderParameters`, so you will need something like this.
```suggestion
		}
	case operatorv1.GCPLoadBalancerProvider:
		if lbs.ProviderParameters.GCP == nil {
			lbs.ProviderParameters.GCP = &operatorv1.GCPLoadBalancerParameters{}
		}
	}

Reply:
Done